cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -61,6 +61,7 @@ InternalFileDecryptor::InternalFileDecryptor(FileDecryptionProperties* propertie
 }
 
 void InternalFileDecryptor::WipeOutDecryptionKeys() {
+  std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
CMIW, this is only called once in `SerializedFile::Close()` so it does not require the lock here.